internal/translator/translate.go

```diff
@@ -109,15 +119,40 @@ func translateRoute(irRoute irv1beta1.Route) (hpv1.Route, []string) {
 	return hpRoute, warnings
 }
 
-func translateRoutes(irRoutes []irv1beta1.Route) ([]hpv1.Route, []string) {
+func translateInclude(irRoute irv1beta1.Route) *hpv1.Include {
+	var hpInclude *hpv1.Include
+
+	if irRoute.Delegate != nil {
+		// If there's a delegation, short-circuit.
+		hpInclude = &hpv1.Include{
+			Conditions: []hpv1.Condition{
+				hpv1.Condition{
+					Prefix: irRoute.Match,
+				},
+			},
+			Name:      irRoute.Delegate.Name,
+			Namespace: irRoute.Delegate.Namespace,
+		}
+	}
+
+	return hpInclude
+}
+
+func translateRoutes(irRoutes []irv1beta1.Route) ([]hpv1.Route, []hpv1.Include, []string) {
 
 	var hpRoutes []hpv1.Route
+	var hpIncludes []hpv1.Include
 	var warnings []string
 	for _, irRoute := range irRoutes {
+		hpInclude := translateInclude(irRoute)
+		if hpInclude != nil {
+			hpIncludes = append(hpIncludes, *hpInclude)
```

Comment:
the dereferencing smell is still here, what is the reason for declaring hpIncludes as a slice of hpv1.Includes, not a slice of *hpv1.Includes?

Reply:
I don't know how to unwrap a slice of *hpv1.Includes to be able to put them into the HTTPProxy spec.

Reply:
(up in the exported function)